src/main/java/at/hannibal2/skyhanni/features/fishing/SeaCreatureFeatures.kt

```diff
@@ -0,0 +1,91 @@
+import at.hannibal2.skyhanni.utils.LocationUtils.distanceToPlayer
+import at.hannibal2.skyhanni.utils.LorenzUtils.editCopy
+import net.minecraft.client.Minecraft
+import net.minecraft.client.entity.EntityOtherPlayerMP
+import net.minecraft.client.gui.FontRenderer
+import net.minecraft.entity.Entity
+import net.minecraft.entity.EntityLivingBase
+import net.minecraft.entity.item.EntityItem
+import net.minecraft.entity.monster.EntityGuardian
+import net.minecraft.entity.monster.EntitySkeleton
+import net.minecraft.entity.monster.EntityZombie
+import net.minecraft.entity.player.EntityPlayer
+import net.minecraft.scoreboard.ScorePlayerTeam
+import net.minecraft.scoreboard.Team.EnumVisible
+import net.minecraftforge.client.event.RenderWorldLastEvent
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import java.awt.Color
+
+
+class SeaCreatureFeatures {
+    private val config get() = SkyHanniMod.feature.fishing
+    private var rareSeaCreatures = listOf<EntityLivingBase>()
+
+    @SubscribeEvent
+    fun onEntityHealthUpdate(event: EntityMaxHealthUpdateEvent) {
+        if (!isEnabled()) return
+        val entity = event.entity as? EntityLivingBase ?: return
+        if (DamageIndicatorManager.isBoss(entity)) return
+
+        val maxHealth = event.maxHealth
+        for (creatureType in RareSeaCreatureType.entries) {
+            if (!creatureType.health.any { entity.hasMaxHealth(it, false, maxHealth) }) continue
+            if (!creatureType.clazz.isInstance(entity)) continue
+
+            if (creatureType.nametag.isNotBlank() && EntityPlayer::class.java.isInstance(entity) && (entity as EntityPlayer).name != creatureType.nametag) {
+                continue
+            }
+
+            rareSeaCreatures = rareSeaCreatures.editCopy { add(entity) }
+            RenderLivingEntityHelper.setEntityColor(entity, LorenzColor.RED.toColor().withAlpha(50))
+            { config.rareSeaCreatureHighlight }
+            RenderLivingEntityHelper.setNoHurtTime(entity) { config.rareSeaCreatureHighlight }
+        }
+    }
+
+    @SubscribeEvent
+    fun onWorldChange(event: LorenzWorldChangeEvent) {
+        rareSeaCreatures = emptyList()
+    }
+
+    @SubscribeEvent
+    fun onRenderEntityOutlines(event: RenderEntityOutlineEvent) {
+        if (isEnabled() && config.rareSeaCreatureHighlight && event.type === RenderEntityOutlineEvent.Type.XRAY) {
+            event.queueEntitiesToOutline(getEntityOutlineColor)
+        }
+    }
+
+    private fun isEnabled() = LorenzUtils.inSkyBlock && !LorenzUtils.inDungeons && !LorenzUtils.inKuudraFight
+
+    private val getEntityOutlineColor: (entity: Entity) -> Int? = { e ->
+        if (EntityLivingBase::class.java.isInstance(e) && e in rareSeaCreatures && e.distanceToPlayer() < 30) {
+            LorenzColor.GREEN.toColor().rgb
+        } else {
+            null
+        }
+    }
+
+    enum class RareSeaCreatureType(val clazz: Class<out EntityLivingBase>, val nametag: String, vararg val health: Int) {
```

Comment:
Consider we could use the repo here instead? 

We already got a "rare" tag in the repo and could add health there instead.

Reply:
see https://github.com/hannibal002/SkyHanni-REPO/blob/ef3cbaaf39f3076565ac621ba834f71352e5e648/constants/SeaCreatures.json#L26

Reply:
Yes, we can simply add the health values in the repo instead.
And you don't need to add the Derpy/runic health values here, just use `entity.hasMaxHealth(health)` instead.

Reply:
We'd have to add the entity class, nametag, and health values there, if thats desirable i could do it though

Reply:
Eventually, yes. Everything in the repo. But since the same data design is used in slayer mini boss feature and somewhere else most likely (can't remember) we would need to port a lot of stuff in the repo. For the moment, it's fine to leave it as is.